android/src/main/java/com/revenuecat/purchases/common/mappers.kt

```diff
@@ -195,7 +195,10 @@ private fun String?.mapPeriodDeprecated(): Map<String, Any?> {
                     "intro_price_period_number_of_units" to 0
                 )
             }
-        }
+        } ?: mapOf(
+            "intro_price_period_unit" to null,
+            "intro_price_period_number_of_units" to null
+        )
```

Comment:
here's the deal, if `PurchasesPeriod.parse` returns `null` because the text is not a Period (i.e. the text is "365" like in the reported crash), `intro_price_period_unit` and `intro_price_period_number_of_units` will be null, but the period itself (like 152 and 159) will be the string "365". So in summary, it's possible to get something like:

```
            "price": 0,
            "priceString": "$0",	            
            "period": "365",
            "cycles" to 1,         
            "intro_price_period_unit": null,
            "intro_price_period_number_of_units": null
	           
```

Which doesn't make sense. Alternatively, we could try to parse the period beforehand, and if it fails, we could nullify the whole object. What do you think??

Reply:
kinda feels weirdly inconsistent to have a "365" in there with null period and number of units. I'd vote for nullifying the whole thing if it's in an unexpected format